test/parseJson.js

```diff
@@ -15,28 +15,28 @@ test('parseJson', function() {
          TypeError,
          'Invalid value\n' +
          '\n' +
-         'parseJson :: TypeRep -> String -> Maybe a\n' +
-         '             ^^^^^^^\n' +
+         'parseJson :: Function -> String -> Maybe b\n' +
+         '             ^^^^^^^^\n' +
          '                1\n' +
          '\n' +
          '1)  "String" :: String\n' +
          '\n' +
-         'The value at position 1 is not a member of ‘TypeRep’.\n');
+         'The value at position 1 is not a member of ‘Function’.\n');
 
-  throws(function() { S.parseJson(Array, [1, 2, 3]); },
+  throws(function() { S.parseJson(S.is(Array), [1, 2, 3]); },
          TypeError,
          'Invalid value\n' +
          '\n' +
-         'parseJson :: TypeRep -> String -> Maybe a\n' +
-         '                        ^^^^^^\n' +
-         '                          1\n' +
+         'parseJson :: Function -> String -> Maybe b\n' +
+         '                         ^^^^^^\n' +
+         '                           1\n' +
          '\n' +
          '1)  [1, 2, 3] :: Array Number, Array FiniteNumber, Array NonZeroFiniteNumber, Array Integer, Array ValidNumber\n' +
          '\n' +
          'The value at position 1 is not a member of ‘String’.\n');
 
-  eq(S.parseJson(Object, '[Invalid JSON]'), S.Nothing);
-  eq(S.parseJson(Array, '{"foo":"bar"}'), S.Nothing);
-  eq(S.parseJson(Array, '["foo","bar"]'), S.Just(['foo', 'bar']));
+  eq(S.parseJson(S.is(Object), '[Invalid JSON]'), S.Nothing);
+  eq(S.parseJson(S.is(Array), '{"foo":"bar"}'), S.Nothing);
+  eq(S.parseJson(S.is(Array), '["foo","bar"]'), S.Just(['foo', 'bar']));
```

Comment:
I'd love to replace the two assertions above with the following:

```javascript
eq(S.parseJson($.test([], $.Array($.String)), '{"foo":"bar"}'), S.Nothing);
eq(S.parseJson($.test([], $.Array($.String)), '[1,2,3,4,5,6]'), S.Nothing);
eq(S.parseJson($.test([], $.Array($.String)), '["foo","bar"]'), S.Just(['foo', 'bar']));
```

Very [aeson][1]-like!

This requires a recent version of sanctuary-def, though. I'm currently working on the pull request to upgrade our sanctuary-def dependency. :)


[1]: https://hackage.haskell.org/package/aeson